lib/internal/fs/promises.js

```diff
@@ -526,17 +524,23 @@ async function readFileHandle(filehandle, options) {
 
   let size = 0;
   let length = 0;
-  if ((statFields[1/* mode */] & S_IFMT) === S_IFREG) {
-    size = statFields[8/* size */];
+  const maxFileSize = 2 ** 31 - 1; // 2 GiB limit for file size
+
+  if ((statFields[1 /* mode */] & S_IFMT) === S_IFREG) {
+    size = statFields[8 /* size */];
+
+    if (size > maxFileSize) {
+      console.warn(
```

Comment:
Please never use `console` in the Node.js code. Please use `process.emitWarning` instead.